students/lee_deitesfeld/lesson04/mailroom.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+
+import sys
+from operator import itemgetter
+
+main_prompt = "\n".join(("Please choose from below mailroom options:",
+          "1 - Manage Donors and Donations",
+          "2 - Create a Report",
+          "3 - Send letters to all donors",
+          "4 - Quit",
+          ">>> "))
+
+
+sub_prompt = "\n".join(("Please make a selection from the following options",
+                        "1 - List the Names of Donors",
+                        "2 - Add a New Donation",
+                        "3 - Return to Main Menu"
+                        ">>> "))
+
+dic_donors = {'donor_db':
+                [
+                {"name": "Bob Jones",
+                 "donations": [63.00, 30.00]
+                },
+                {"name": "Sue Smith",
+                 "donations": [50.00]
+                },
+                {"name": "Joe Grimes",
+                 "donations": [102.00, 45.00, 70.00]
+                },
+                {"name": "Andrea Funk",
+                 "donations": [21.00, 21.00]
+                },
+                {"name": "Mickey Mouse",
+                 "donations": [500.00, 250.00, 100.00]
+                }
+                ]}
```

Comment:
even though this is a dictionary, it isn't exactly the dictionary ideal for this data structure.

The benefit of using dict for donors data structure is its efficiency in look ups by key - in our case donor's name makes a perfect key since we are usually searching donors by name to see if donations should be updated or added.

In the above structure you are still dealing with a list of donors thus not really satisfying the requirement.

Reply:
so a better data structure here would be:
```
donors = {"Jeff Bezos": [993, 999, 232], "Tom Jerry": [33, 244]}
```